### PR TITLE
macros for OHPC-style builds of multiple python flavored packages from a single spec

### PR DESCRIPTION
**Pull request by crbaird**
This PR adds the notion of a python family similar to our compiler and mpi families to allow a single spec file to produce multiple rpms for both python2.7 and python3.4. It defaults to python3, and a _link file in OBS can be used to use python2 as an alternative.

Signed-off-by: Reese Baird <reese.baird@intel.com>

### CODE REVIEW

File: components/OHPC_macros [hunk 1]
```diff
@@ -35,9 +35,9 @@
 DocDir:    %{OHPC_PUB}/doc/contrib
 
 %if 0%{?rhel} == 7
-	# OBS define's rhel_version to 700 for RHEL, in case we are
-	# not running in OBS
-	%global rhel_version 700
+    # OBS define's rhel_version to 700 for RHEL, in case we are
+    # not running in OBS
+    %global rhel_version 700
```
Comment (adrianreber) [inline components/OHPC_macros]: These whitespace changes seem unrelated to the actual python3 change.

File: components/OHPC_macros [hunk 2]
```diff
@@ -115,9 +116,32 @@ Requires:      openmpi3-%{compiler_family}%{PROJ_DELIM}
 %endif
 %endif
 
+# Python dependencies and macros
+%if 0%{?ohpc_python_dependent} == 1
+%if "%{python_family}" == "python3"
+%global __python %__python3
```
Comment (adrianreber) [inline components/OHPC_macros]: Where is %__python3 defined?

File: components/OHPC_macros [hunk 3]
```diff
@@ -115,9 +116,32 @@ Requires:      openmpi3-%{compiler_family}%{PROJ_DELIM}
...
-	. %{OHPC_ADMIN}/ohpc/OHPC_setup_compiler %{compiler_family} \
-	%if 0%{?ohpc_mpi_dependent} == 1 \
-		. %{OHPC_ADMIN}/ohpc/OHPC_setup_mpi %{mpi_family} \
-	%endif \
+    . %{OHPC_ADMIN}/ohpc/OHPC_setup_compiler %{compiler_family} \
+    %if 0%{?ohpc_mpi_dependent} == 1 \
+        . %{OHPC_ADMIN}/ohpc/OHPC_setup_mpi %{mpi_family} \
+    %endif \
 }
+
```
Comment (adrianreber) [inline components/OHPC_macros]: More whitespace changes.

File: components/OHPC_macros (hunk 2 above)
Comment (crbaird) [reply components/OHPC_macros]: it's in EPEL's python3-rpm-macros, a requirement of python34-devel

**Comment by adrianreber** [comment]
Thanks for the clarification. :+1: 

One idea would be to move the BuildRequires to this file, so that we can just set ohpc_python_dependent to 1 and all BuildRequires and Requires are automatically set like with the compilers and MPI. Just an idea. Maybe even your intel compiler python34-build-patch requirement could be in the central macro file.
